prototypes/ScheduledMerges.hs

```diff
@@ -561,78 +821,81 @@ data EventDetail =
          runsAtLevel   :: Int
        }
      | MergeStartedEvent {
-         mergePolicy   :: MergePolicy,
-         mergeLast     :: MergeLastLevel,
-         mergeDebt     :: MergeDebt,
+         mergePolicy   :: Maybe MergePolicy,
+         mergeType     :: MergeType,
+         mergeDebt     :: Int,
          mergeCost     :: Int,
          mergeRunsSize :: [Int]
        }
      | MergeCompletedEvent {
-         mergePolicy :: MergePolicy,
-         mergeLast   :: MergeLastLevel,
+         mergePolicy :: Maybe MergePolicy,
+         mergeType   :: MergeType,
          mergeSize   :: Int
```

Comment:
Yes, that's one of the questions to answer around tracing. It's a little complicated, since level merges (merges created by inserting values) also become part of a union and might get completed through `supplyUnionCredits`. Should these then have a different CompletedEvent than merges created through `union`? Seems a bit awkward to have to make this distinction (especially in case we don't want the "[tweak PendingMerge representation](https://github.com/IntersectMBO/lsm-tree/pull/477/commits/ac5c036904b3f53b0f257dd62d30d1c0f6128878)" change, which currently tracks the origin of a merge in a way).